frontend/app/components/ui/Toast.tsx

```diff
@@ -0,0 +1,42 @@
+"use client";
+import { ReactNode, forwardRef, useImperativeHandle, useState } from "react";
+import * as ToastPrimitive from "@radix-ui/react-toast";
+import Button from "./Button";
+
+interface ToastProps extends ToastPrimitive.ToastProps {
+  children?: ReactNode;
+}
+
+export const Toast = forwardRef(
+  ({ children, ...props }: ToastProps, forwardedRef) => {
+    const [count, setCount] = useState(0);
+
+    useImperativeHandle(forwardedRef, () => ({
+      publish: () => setCount((count) => count + 1),
+    }));
+
+    return (
+      <>
+        <ToastPrimitive.Provider duration={Infinity}>
+          {Array.from({ length: count }).map((_, index) => (
```

Comment:
Some remarks:

1. As new toasts are added, all indices (and thus keys) change, causing unnecessary re-rendering of all toasts, leading to performance inefficiencies.

2. If in the future you decide to allow toasts to be removed individually, using indices as keys could lead to incorrect behavior or bugs due to shifted indices.

3. React uses keys for diffing and deciding which components to update, thus if the count changes, using indices could lead to incorrect diff results, causing potential issues with state updates.

Proposition:

```ts
useImperativeHandle(forwardedRef, () => ({
  publish: () => setToasts((toasts) => [...toasts, { id: generateUniqueID(), content: children }]),
}));
```
```ts
 key={toast.id}
```

Reply:
Toasts now get different ids when published which are generated by this function in `@/lib/utils.ts` file.

```ts
export function generateUniqueId() {
  const timestamp = Date.now();
  const random = Math.floor(Math.random() * 10000);
  return `${timestamp}-${random}`;
}
```